### PR TITLE
support incremental training in XGBoost

### PR DESCRIPTION
**Pull request by sneaxiy**
Extend `TO TRAIN` syntax to support incremental training. The SQL is like:

```sql
SELECT * FROM ...
TO TRAIN ...
WITH ...
LABEL ...
[USING pre_trained_model_name_to_load]
INTO new_trained_model_name_to_save;
```

### CODE REVIEW

File: pkg/sql/codegen/pai/codegen.go [hunk 1]
```diff
@@ -122,20 +122,24 @@ func Train(ir *ir.TrainStmt, session *pb.Session, tarball, paramsFile, modelName
 			return
 		}
 	} else if strings.HasPrefix(strings.ToLower(ir.Estimator), "xgboost") {
-		ossURI := OSSModelURL(ossModelPath)
-		if code, e = xgboost.DistTrain(ir, session, cc.Worker.Count, ossURI); e != nil {
+		ossURIToSave := OSSModelURL(ossModelPathToSave)
+		ossURIToLoadPreTrainedModel := ""
```
Comment (typhoonzero) [inline pkg/sql/codegen/pai/codegen.go]: Simplify to `ossURIToLoad`?

Comment (typhoonzero) [reply pkg/sql/codegen/pai/codegen.go]: Check if `ossModelPathToSave == ossModelPathToLoadPreTrained`?

Comment (sneaxiy) [reply pkg/sql/codegen/pai/codegen.go]: > Simplify to ossURIToLoad?

Done.

> Check if ossModelPathToSave == ossModelPathToLoadPreTrained?

There is no need to check. We can support both `save_path == load_path` and `save_path != load_path`.

File: pkg/parser/extended_syntax_parser.y [hunk 2]
```diff
@@ -76,11 +76,12 @@ type StandardSelect struct {
...
-	Estimator  string
-	TrainAttrs Attributes
-	Columns    columnClause
-	Label      string
-	Save       string
+	Estimator       string
+	TrainAttrs      Attributes
+	Columns         columnClause
+	Label           string
+	PreTrainedModel string
```
Comment (typhoonzero) [inline pkg/parser/extended_syntax_parser.y]: parser and IR represents the parsed SQL statement information, `PreTrainedModel` is more likely to be how this `USING` field will be used, not representing what the parse the field is. Maybe change it to `TrainUsing` ?

**Review by typhoonzero — DISMISSED** [review]
LGTM with one naming issue.

File: pkg/sql/codegen/pai/codegen.go (hunk 1 above)
Comment (typhoonzero) [reply pkg/sql/codegen/pai/codegen.go]: I see.

File: pkg/parser/extended_syntax_parser.y (hunk 2 above)
Comment (sneaxiy) [reply pkg/parser/extended_syntax_parser.y]: Sounds good. Done.

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**